### PR TITLE
LPD-42742 Style book unrestricted File Upload Vulnerability

### PR DESCRIPTION
**Pull request by thiago-buarqque**
LPD: https://liferay.atlassian.net/browse/LPD-42742
LPP: https://liferay.atlassian.net/browse/LPP-56829

# Issue
Style book preview upload does not validate file type, which allows uploading and downloading any file.

### CODE REVIEW

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java [hunk 1]
```diff
@@ -92,6 +93,60 @@ public void setUp() throws Exception {
 		_themeDisplay.setUser(TestPropsValues.getUser());
 	}
 
+	@Test(expected = NoSuchFileEntryException.class)
+	public void testInvalidStyleBookEntryPreview() throws Exception {
+		MockLiferayPortletActionRequest mockLiferayPortletActionRequest =
```
Comment (algobob) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java]: You could move this initialization after `File fileEntry = ...` where its first usage happens

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java [hunk 2]
```diff
@@ -92,6 +93,60 @@ public void setUp() throws Exception {
...
+			new MockLiferayPortletActionRequest();
+
+		StyleBookEntry styleBookEntry =
+			_styleBookEntryLocalService.addStyleBookEntry(
+				null, TestPropsValues.getUserId(), _group.getGroupId(), false,
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				_serviceContext);
+
+		Class<?> clazz = getClass();
```
Comment (algobob) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java:103]: Move this to line before `StyleBookEntry styleBookEntry = ...`

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java [hunk 3]
```diff
@@ -92,6 +93,60 @@ public void setUp() throws Exception {
...
+
+		Assert.assertTrue(
+			SessionErrors.contains(
+				mockLiferayPortletActionRequest, "invalidThumbnailFormat"));
+
+		PortletFileRepositoryUtil.getPortletFileEntry(
+			_group.getGroupId(), _repository.getDlFolderId(),
+			_getFileName(
+				fileEntry.getExtension(),
+				styleBookEntry.getStyleBookEntryId()));
```
Comment (algobob) [inline modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java]: Usually assert instructions are the last part of the test. Is there a usage of this one? It seems it returns a file entry but no one use it

Comment (thiago-buarqque) [reply modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java]: @algobob  This call will throw `NoSuchFileEntryException`, it is stated at line 96.

File: modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java (hunk 1 above)
Comment (thiago-buarqque) [reply modules/apps/style-book/style-book-web-test/src/testIntegration/java/com/liferay/style/book/web/internal/portlet/action/test/UpdateStyleBookEntryPreviewMVCActionCommandTest.java]: This is done in the last commit

**Review by ethib137 — APPROVED** [review]
LGTM... Thanks @thiago-buarqque 

If @algobob also approves it feel free to send to BPM.

**Comment by algobob** [comment]
LGTM :rocket: 

For next PR's, what do you think about split refactoring commits in two? One for rename only and another for move code around. It would be easier to read.

**Comment by thiago-buarqque** [comment]
Resent to BPM https://github.com/liferay-bpm/liferay-portal/pull/3908

**Pull request closed without merging**